File: domain/src/main/java/com/study/platform/global/idempotency/IdempotencyObjectStoragePort.java
package com.study.platform.global.idempotency;

import java.time.Duration;

public interface IdempotencyObjectStoragePort {

    Object get(String key);

    boolean setIfAbsent(String key, String marker, Duration ttl);

    void set(String key, Object value, Duration ttl);

    void delete(String key);
}

File: domain/src/main/kotlin/com/study/platform/global/idempotency/IdempotencyObjectStoragePort.kt
package com.study.platform.global.idempotency

import java.time.Duration

interface IdempotencyObjectStoragePort {
    fun get(key: String): Any?
    fun setIfAbsent(key: String, marker: String, ttl: Duration): Boolean
    fun set(key: String, value: Any, ttl: Duration)
    fun delete(key: String)
}
